### ISSUE TITLE
UI improvements from June 2 testing

### ISSUE THREAD

**Issue by emilyeros** [sharedstreets/curb-wheel#85]
In priority order...

### Important-easy
- [x] Survey screen: use bigger font on wheel counter number. It was really helpful to keep an eye on the counter since it was more precise than the progress bar
- [ ] Survey screen: Add visual indication when there are more curb zones or buttons below (when not everything is visible on the screen and scrolling is required). Maybe this is just a scroll bar.
- [x] Survey screen: Improve interaction with active curb zones: Make whole button responsive instead of just the gear icon, and consider switching the gear icon to a "hamburger"
- [x] Survey screen: Make it more obvious when a curb zone is open vs closed (fade... color coding?... icon... something else?)
- [x] Survey screen: Selecting a label for the span/position: it's a little hard to pick one. Consider changing to 2x column layout, or changing to a "scrolly dropdown"
- [x] Survey screen: Give feedback that an image was taken. Basic path: add number counter. More involved path: add diamond to span. (Gets a little trickier for positions since they already have a diamond on the progress bar)

### Important-harder
- [x] Overworld map: Color-code streets to indicate whether a street is fully surveyed (both sides) or not yet fully surveyed (0 or 1 side surveyed)
- [x] Overworld street selector: decouple side of street from direction being walked. In UI, choose side of street, then choose direction (could be map selection or two different toggle buttons). Related: When selecting street side, indicate to user if one/both sides of the street have already been mapped.
- [x] Overworld map: Help users orient themselves with puck. (morgan/kevin investigating location permissions and creating support for puck (in a separate issue)

### Tabled for now
- Color coding for curb zones (let's make the improvements above first, then come back to this)
- Visualization improvements for T intersections (when on the unsegmented side of the street, it's hard to know where the block segmentation is)

**Comment by peterqliu** [comment]
Most of the first section is implemented, at https://github.com/sharedstreets/curb-wheel/pull/92. 

>Survey screen: Give feedback that an image was taken

this is contingent on handling https://github.com/sharedstreets/curb-wheel/pull/91, which actually handles the image capture

**Comment by peterqliu** [comment]
#92 is merged with all the `Important-easy` items except the scroll bar -- mobile browsers are pretty prescriptive about how those appear, so I'm working on a custom implementation of that

>decouple side of street from direction being walked. In UI, choose side of street, then choose direction (could be map selection or two different toggle buttons).

@emilyeros do we have `direction` as a property in the data model, if we're decoupling direction from street side? If not, we could flip it on the client before uploading the survey, so that the data is always in the "forward" direction.

**Comment by emilyeros** [comment]
Direction is baked into the reference IDs (https://github.com/sharedstreets/curb-wheel/blob/master/data-model.md#street-linear-references). My understanding: the `A` side of the street does have a direction (forward) and the toggle would be a way of opting to use the back reference for that ID. I don't think we've explicitly surfaced the `forward` property in the data though.

@morganherlocker / @kpwebb could confirm all this? 

If this properly isn't explicitly in there and we do need it, then I think it would be simple to add this property back in (or add these features into the graph?).

**Comment by peterqliu** [comment]
>Direction is baked into the reference IDs

@emilyeros hmm I'm getting more confused. Ref ID's have "direction" derived from  car traffic, but we're talking about which way the surveyor will walk, given a specific stretch of curb: from the Arby's to the Starbucks, or Starbucks to the Arby's. A one-way street with one ref ID will still have **two** "directions" of surveying for each curb: their contents will be identical but palindromic.

My question is whether we want to

- upload surveys with these contrary measurements, along with a property recording the walking direction, or
- decide on a canonical "forward walking direction", and have the client coerce "backward" surveys into the forward format before uploading

the latter seems more straightforward and simplifies the data model.

**Comment by peterqliu** [comment]
https://github.com/sharedstreets/curb-wheel/pull/95 for puck and curb selection items on the map portion

**Comment by emilyeros** [comment]
Last remaining item is the color-coded overworld map. Current status: needs a status field for each street reference on the backend. Peter added a hacky workaround that stores the street state in the web browser, but it doesn't persist if the page is refreshed.

**Comment by emilyeros** [comment]
done
